### ISSUE TITLE
Recent 4.1.0-SNAPSHOTS break at runtime.

### ISSUE THREAD

**Issue by nealsanche** [mapbox/mapbox-gl-native#5135]
**Platform:Android API 23**
**Mapbox SDK version:4.1.0-SNAPSHOT**

### Steps to trigger behavior

 1. Compile and try to start an app.
 2. App dies trying to display the MapView

### Expected behavior

Map would display.

### Actual behavior

Map doesn't display, but instead the app dies with the following exception:

```
Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZN6mapbox9geojsonvt7Convert11projectRingERKNSt6__ndk16vectorINS0_6LonLatENS2_9allocatorIS4_EEEEd" referenced by "/data/app/com.acme.firebans/lib/arm64/libmapbox-gl.so"...
```

**Comment by nealsanche** [comment]
I'll get a better stack trace. I tried compiling against mapbox 4.0.0, and the app is working fine.

**Comment by nealsanche** [comment]
```
05-25 12:08:12.395 22736-22736/com.acme.firebans E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                         Process: com.acme.firebans, PID: 22736
                                                                                         java.lang.RuntimeException: Unable to start activity ComponentInfo{com.acme.firebans/com.acme.firebans.activities.MainActivity}: android.view.InflateException: Binary XML file line #42: Binary XML file line #42: Error inflating class com.mapbox.mapboxsdk.maps.MapView
                                                                                             at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2625)
                                                                                             at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2686)
                                                                                             at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                                             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1440)
                                                                                             at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                             at android.os.Looper.loop(Looper.java:154)
                                                                                             at android.app.ActivityThread.main(ActivityThread.java:5969)
                                                                                             at java.lang.reflect.Method.invoke(Native Method)
                                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:801)
                                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:691)
                                                                                          Caused by: android.view.InflateException: Binary XML file line #42: Binary XML file line #42: Error inflating class com.mapbox.mapboxsdk.maps.MapView
                                                                                          Caused by: android.view.InflateException: Binary XML file line #42: Error inflating class com.mapbox.mapboxsdk.maps.MapView
                                                                                          Caused by: java.lang.reflect.InvocationTargetException
                                                                                             at java.lang.reflect.Constructor.newInstance0(Native Method)
                                                                                             at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
                                                                                             at android.view.LayoutInflater.createView(LayoutInflater.java:645)
                                                                                             at uk.co.chrisjenx.calligraphy.CalligraphyLayoutInflater.createCustomViewInternal(CalligraphyLayoutInflater.java:211)
                                                                                             at uk.co.chrisjenx.calligraphy.CalligraphyLayoutInflater.access$000(CalligraphyLayoutInflater.java:20)
                                                                                             at uk.co.chrisjenx.calligraphy.CalligraphyLayoutInflater$PrivateWrapperFactory2.onCreateView(CalligraphyLayoutInflater.java:302)
                                                                                             at android.view.LayoutInflater$FactoryMerger.onCreateView(LayoutInflater.java:189)
                                                                                             at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:777)
                                                                                             at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
                                                                                             at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
                                                                                             at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
                                                                                             at android.view.LayoutInflater.rInflate(LayoutInflater.java:861)
                                                                                             at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
                                                                                             at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
                                                                                             at uk.co.chrisjenx.calligraphy.CalligraphyLayoutInflater.inflate(CalligraphyLayoutInflater.java:60)
                                                                                             at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
                                                                                             at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
                                                                                             at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:280)
                                                                                             at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
                                                                                             at android.databinding.DataBindingUtil.setContentView(DataBindingUtil.java:276)
                                                                                             at android.databinding.DataBindingUtil.setContentView(DataBindingUtil.java:261)
                                                                                             at com.robotsandpencils.goafirebansandroid.activities.MainActivity.onCreate(MainActivity.java:107)
                                                                                             at android.app.Activity.performCreate(Activity.java:6669)
                                                                                             at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                                                             at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2578)
                                                                                             at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2686)
                                                                                             at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                                                             at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1440)
                                                                                             at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                             at android.os.Looper.loop(Looper.java:154)
                                                                                             at android.app.ActivityThread.main(ActivityThread.java:5969)
                                                                                             at java.lang.reflect.Method.invoke(Native Method)
                                                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:801)
                                                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:691)
05-25 12:08:12.395 22736-22736/com.acme.firebans E/AndroidRuntime: Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZN6mapbox9geojsonvt7Convert11projectRingERKNSt6__ndk16vectorINS0_6LonLatENS2_9allocatorIS4_EEEEd" referenced by "/data/app/com.acme.firebans/lib/arm64/libmapbox-gl.so"...
                                                                                             at java.lang.Runtime.loadLibrary0(Runtime.java:977)
                                                                                             at java.lang.System.loadLibrary(System.java:1530)
                                                                                             at com.mapbox.mapboxsdk.maps.NativeMapView.<clinit>(NativeMapView.java:44)
                                                                                             at com.mapbox.mapboxsdk.maps.MapView.initialize(MapView.java:203)
                                                                                             at com.mapbox.mapboxsdk.maps.MapView.<init>(MapView.java:166)
                                                                                             	... 34 more
05-25 12:08:13.830 22906-22906/com.acme.firebans E/GMPM: GoogleService failed to initialize, status: 10, Missing an expected resource: 'R.string.google_app_id' for initializing Google services.  Possible causes are missing google-services.json or com.google.gms.google-services gradle plugin.
05-25 12:08:13.830 22906-22906/com.acme.firebans E/GMPM: Scheduler not set. Not logging error/warn.
05-25 12:08:13.895 22906-22937/com.acme.firebans E/GMPM: Uploading is not possible. App measurement disabled
```

**Comment by Schumi09** [comment]
Hi there,

it's (hopefully) already fixed in #5108. However, the today's snapshot build has failed (see #5126).

**Comment by bleege** [comment]
@nealsanche @Schumi09 Yep, this related to some changes in the Core GL process that we've identified in #5126.  These changes caused the MIPS ABI not to build (the remaining 5 were fine) so this prevented us from publishing a new SNAPSHOT since Monday night's broken NDK build.  We've found the [problem now and are taking steps to remedy it](https://github.com/mapbox/mapbox-gl-native/issues/5126#issuecomment-221647573).  We should have a new `4.1.0-SNAPSHOT` available in a few hours.  Thanks!

**Comment by bleege** [comment]
@nealsanche @Schumi09 Just wanted to let you know that I just published a new `4.1.0-SNAPSHOT` (Number 36) and [it's good to go](https://github.com/mapbox/mapbox-gl-native/issues/5126#issuecomment-221688182).  Thanks for your patience!

**Comment by nealsanche** [comment]
Thanks. I'll have a look.

On Wed, May 25, 2016, 2:57 PM Brad Leege <notifications@github.com> wrote:

> @nealsanche <https://github.com/nealsanche> @Schumi09
> <https://github.com/Schumi09> Just wanted to let you know that I just
> published a new 4.1.0-SNAPSHOT (Number 36) and it's good to go
> <https://github.com/mapbox/mapbox-gl-native/issues/5126#issuecomment-221688182>.
> Thanks for your patience!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mapbox/mapbox-gl-native/issues/5135#issuecomment-221688569>
>

-- 
Robots and Pencils Inc | 1507 14th Ave SW Calgary AB T3C 0W4
You can unsubscribe from the Robots and Pencils mailing lists by clicking 
here. <https://robotsandpencils.com/contact/unsubscribe.html>
